src/flamegraph/mod.rs

```diff
@@ -434,16 +441,21 @@ where
     let cache_a_end = Event::End(BytesEnd::borrowed(b"a"));
 
     // create frames container
-    if let Event::Start(ref mut g) = cache_g {
-        g.extend_attributes(std::iter::once(("id", "frames")));
-    }
-    svg.write_event(&cache_g)?;
+    let container_x = format!("{}", XPAD);
+    let container_width = format!("{}", image_width as usize - XPAD - XPAD);
+    svg.write_event(Event::Start(
+        BytesStart::borrowed_name(b"svg").with_attributes(vec![
```

Comment:
Don't we already write out `<svg>` in `write_prelude` above? Should this be `<g>` instead?

Reply:
I changed the "frames" container to `<svg>` to make it easier to maintain fixed padding on the left and right when resizing.

Reply:
Is an `<svg>` even allowed to contain additional `<svg>` elements? Why can't it continue to be a `<g>` that you then set the position/width on?

Reply:
You would think that'd work, but it doesn't. The `x` and `width` attributes can't be used on a `<g>`. Working with SVG is frustrating because many things don't work like you'd think. This could probably be done much more easily in html.